tests/data/l7irish/data.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import hashlib
+import os
+import shutil
+from typing import Dict, List, Union
+
+import numpy as np
+import rasterio
+from rasterio import Affine
+from rasterio.crs import CRS
+
+SIZE = 36
+
+np.random.seed(0)
+
+FILENAME_HIERARCHY = Union[Dict[str, "FILENAME_HIERARCHY"], List[str]]
+
+bands = [
+    "B10.TIF",
+    "B20.TIF",
+    "B30.TIF",
+    "B40.TIF",
+    "B50.TIF",
+    "B61.TIF",
+    "B62.TIF",
+    "B70.TIF",
+    "B80.TIF",
+]
+
+filenames: FILENAME_HIERARCHY = {
+    "austral": {"p226_r98": [], "p227_r98": [], "p231_r93_2": []},
+    "boreal": {"p2_r27": [], "p143_r21_3": []},
+}
+prefixes = [
+    "L71226098_09820011112",
+    "L71227098_09820011103",
+    "L71231093_09320010507",
+    "L71002027_02720010604",
+    "L71143021_02120010803",
+]
+
+for land_type, patches in filenames.items():
+    for patch in patches:
+        p, r = patch.split("_r")
+        if r.find("_") == -1:
+            r = r.zfill(3)
+        else:
+            r = r[: r.find("_")].zfill(3)
+        key = p[1:] + r
```

Comment:
Does p need zero padding too? Why not use:
```suggestion
        path, row = patch.split("_")[:2]
        path = path[1:].zfill(3)
        row = row[1:].zfill(3)
        key = path + row
```